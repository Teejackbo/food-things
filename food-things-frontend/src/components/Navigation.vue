<template>
  <nav>
    <section class="nav-left">
      <router-link :to="{ name: 'home' }">Feed</router-link>
    </section>
    <section class="nav-right">
      <router-link :to="{ name: 'create-food' }">Create Food</router-link>
      <router-link v-if="user.id" :to="{ name: 'logout' }">Logout</router-link>
      <router-link v-else :to="{ name: 'login' }">Login</router-link>
      <router-link v-if="!user.id" :to="{ name: 'sign-up' }">Sign Up</router-link>
    </section>
  </nav>
</template>

<script>
import { mapState } from 'vuex'

export default {
  name: 'Navigation',
  computed: mapState({
    user: state => state.user.user,
  }),
}
</script>

<style lang="scss" scoped>
@import '../style/variables';

nav {
  align-items: center;
  background: linear-gradient(to bottom, $primary, lighten($primary, 4%));
  box-shadow: 0px 3px 8px $shadow;
  display: flex;
  font-family: $font-body;
  justify-content: space-between;
  margin-bottom: 20px;
  padding: 1.5em 2em;
}

a {
  color: $white;
  font-size: 18px;
  font-weight: 600;
  letter-spacing: 0.6px;
  margin: 0 1em;
  text-decoration: none;
  text-shadow: 0 2px 5px $shadow;
  text-transform: uppercase;
  transition: color 0.3s, text-shadow 0.3s;

  &:hover {
    color: $white-lightest;
    text-shadow: 0 3px 8px $shadow;
  }
}
</style>
