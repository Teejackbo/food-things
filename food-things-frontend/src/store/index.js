import Vue from 'vue'
import Vuex from 'vuex'

import user from '@/store/modules/user'
import alerts from '@/store/modules/alerts'
import food from '@/store/modules/food'

Vue.use(Vuex)

export default new Vuex.Store({
  modules: {
    user,
    alerts,
    food,
  },
  state: {},
  mutations: {},
  actions: {},
})
