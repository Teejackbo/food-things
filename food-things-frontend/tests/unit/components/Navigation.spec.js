import { shallowMount } from '@vue/test-utils'
import Vue from 'vue'
import VueRouter from 'vue-router'
import Navigation from '@/components/Navigation'
import $store from '@/store'

Vue.use(VueRouter)

function getWrapper() {
  return shallowMount(Navigation, {
    mocks: {
      $store,
    },
  })
}

describe('Navigation', () => {
  it('Renders a nav element.', () => {
    const wrapper = getWrapper()

    expect(wrapper.contains('nav')).toBe(true)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('Should contain two sections.', () => {
    const wrapper = getWrapper()

    expect(wrapper.findAll('section')).toHaveLength(2)
  })
})
