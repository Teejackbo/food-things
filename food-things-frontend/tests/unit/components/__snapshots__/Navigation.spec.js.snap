// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Renders a nav element. 1`] = `
<nav>
  <section
    class="nav-left"
  >
    <vuecomponent-stub
      to="[object Object]"
    />
  </section>
   
  <section
    class="nav-right"
  >
    <vuecomponent-stub
      to="[object Object]"
    />
     
    <vuecomponent-stub
      to="[object Object]"
    />
     
    <vuecomponent-stub
      to="[object Object]"
    />
  </section>
</nav>
`;
